source/slang/slang-ir-dce.cpp

```diff
@@ -512,6 +512,184 @@ bool trimOptimizableTypes(IRModule* module)
     return changed;
 }
 
+static bool eliminateLoadStorePairsInFunc(IRFunc* func, HashSet<IRFunc*>& processedFuncs)
+{
+    // Avoid infinite recursion by tracking processed functions
+    if (processedFuncs.contains(func))
+        return false;
+
+    processedFuncs.add(func);
+
+    bool changed = false;
+    HashSet<IRInst*> toRemove;
+
+    for (auto block : func->getBlocks())
+    {
+        for (auto blockInst : block->getChildren())
+        {
+            // First, recursively process any function calls in this block
```

Comment:
Why is this recursion necessary? Can't we push the function into a work list and process it later?

Also the function is named eliminateXxxInFunc, that suggests that the function is operating locally on a single function and not making changes beyond the function. We should use a work list and iterate on functions outside this eliminateLoadStorePairInFunc function.